grails-app/controllers/org/pih/warehouse/outbound/PackingListController.groovy

```diff
@@ -0,0 +1,22 @@
+package org.pih.warehouse.outbound
+
+import grails.converters.JSON
+import org.pih.warehouse.core.UploadService
+import org.pih.warehouse.importer.DataImporter
+import org.pih.warehouse.importer.ImportDataCommand
+import org.pih.warehouse.importer.PackingListExcelImporter
+import org.springframework.web.multipart.MultipartFile
+
+class PackingListController {
+
+    UploadService uploadService
+
+    def importPackingList(ImportDataCommand command) {
+        MultipartFile importFile = command.importFile
+        File localFile = uploadService.createLocalFile(importFile.originalFilename)
+        importFile.transferTo(localFile)
+        DataImporter packingListImporter = new PackingListExcelImporter(localFile.absolutePath)
+
+        render([data: packingListImporter.toJson() ] as JSON)
```

Comment:
Interesting idea to add the toJson to the importer object. I would have expected this to be the resposibility of the command object.  I need to think about this more, but it feels like the importer should be responsible for parsing the data and the command object should be responsible for rendering the data. 

I could be wrong. Maybe the importer is a glorified command object itself? Anyone else have any thoughts?

In general, it feels like we'll want to rememeber this when we refactor import/export since we want to be able to import and export in different formats. I'm not sure if the importer / exporter should be responsible for that. Instead we delegate that to a class that implements a Strategy pattern (or maybe Adapter pattern). 

https://pihemr.atlassian.net/browse/OBIG-4

Reply:
As for the importer itself, I was just following the pattern we already have, as it was painful to make it work. I agree there is a bit of mess with the exporters/importers and this should be refactored at some point, but answering your question I'd say we could treat the importer class as the command, since it contains the data